### ISSUE TITLE
Kubeflow v1.2 Rebasing

### ISSUE THREAD

**Issue by wg102** [StatCan/kubeflow#34]
This is to keep track of the steps for the rebasing, and the conflicts that arise.

**Comment by saffaalvi** [comment]
Steps taken for [stc-master-v1.2-merge branch](https://github.com/StatCan/kubeflow/tree/stc-master-v1.2):

**Merge** (keeps our commits and history, adds upstream changes as one big merge commit with one author aka me) 
1. Clone whole repository (in this case, StatCan/kubeflow) 
2. Go into repo locally 
3. git remote add upstream https://github.com/kubeflow/kubeflow.git 
4. git remote –v (list current configured remote) 
5. git checkout stc-master 
6. git checkout –b <new-branch-name> 
7. git fetch upstream 
8. git merge upstream/v1.2-branch 
9. Resolve conflicts -> save -> stage changes 
10. git commit 
11. git push --set-upstream origin <new-branch-name> 

If you want to **rebase**, brings in all commits from upstream, but loses **our** commit history (adds all commits on one date and adds me as author to everything), change `merge` in step 8 to `rebase`. Branch: https://github.com/StatCan/kubeflow/tree/stc-master-v1.2-merge

**Comment by saffaalvi** [comment]
After discussing with the team, using `git rebase` is a better option since it will preserve both the kubflow upstream commit history as well as ours.

**Comment by saffaalvi** [comment]
Branch https://github.com/StatCan/kubeflow/tree/v1.2-branch has been rebased to the v1.2-branch upstream.
Conflicts that came up while rebasing, which we tried to resolve but are going to take another look:

**Central Dashboard now as configmap**
Upstream commit: https://github.com/kubeflow/kubeflow/commit/ec7e01a28bf5339f784fdfddcb7bb0bcb46ce352#diff-a29b81659d6ca88b1df24f77d04886f8eb04631f08a34e41da53e6bb273e3be0
Will have to go back and integrate these changes with our links, documentation, and restructure.

**i18n (en/fr)**
Our commit: https://github.com/StatCan/kubeflow/commit/f85a597e5040ab518b44da81b7582d5c0252f492
Upstream has no translations at all, so we have to fit this in with their new configmap.

**Security Update**
Our commit: https://github.com/StatCan/kubeflow/commit/30116175161aace30ffc3b8007d194368ef9ae40#diff-a29b81659d6ca88b1df24f77d04886f8eb04631f08a34e41da53e6bb273e3be0
We changed for security vulnerabilities:
`coreAPI: k8s.Core_v1Api;` to `k8s.CoreV1Api;`
`customObjectsAPI: k8s.Custom_objectsApi;` to `k8s.CustomObjectsApi;`
Upstream has not updated this yet.

**Comment by wg102** [comment]
We need to find a way to **test** central dashboard. Since the links are now coming from configmap, and it's an API call '`/dashboard-links`' the local npm run dev now shows an almost empty page.  
There are 4 errors like these:

> Failed to load resource: the server responded with a status of 504 (Gateway Timeout)

**Comment by saffaalvi** [comment]
We ran into an issue where there was an error retrieving the pipelines. Zach fixed the error in this commit: https://github.com/StatCan/kubeflow/commit/5a07fb421e54a110374c8df9aebc9a1d43e188b8 and Wendy fixed the pipeline tests here: https://github.com/StatCan/kubeflow/commit/77c965c7b1999756ae224909666529bd291a5da5. Not sure if this is a problem upstream but they didn't seem to have any visible fix for it.

**Comment by wg102** [comment]
We ran into the issue of the configmap. The menu links, quick links and documentation links now come from a centraldashboard-links config map which we were not able to customize to our docs directly from changing the file. It seems like a different resource when looking at the Network tab. The resource it uses is viewable with this command : `kubectl -n kubeflow get configmap centraldashboard-links-config` This will need to be addressed. For now, we removed the code that was using it and kept the arrays that were previously used.
